lib/src/metta/runner/mod.rs

```diff
@@ -241,6 +241,14 @@ impl Metta {
         self.0.settings.borrow().get(key.into()).map(|a| a.to_string())
     }
 
+    /// Runs the MeTTa code in the specified file
+    pub fn run_from_file<P: AsRef<Path>>(&self, path: P) -> Result<Vec<Vec<Atom>>, String> {
```

Comment:
I would suggest making this method private. Each public method increases the probability of the compatibility issues which happens on the our side. I would introduce an `SExprParser` constructor which is initialized by the instance of the `Path`.
```suggestion
    fn run_from_file<P: AsRef<Path>>(&self, path: P) -> Result<Vec<Vec<Atom>>, String> {
```

Reply:
One of the ideas behind `SExprParser` is that programmer can implement the parser for another MeTTa syntax and use it with the runner. It requires additional work indeed, for example we need to abstract the parser API and remove places where `SExprParser` instances are created inside the runner.

Reply:
That makes sense.

I thought about adding a convenience to SExprParser to ingest a file, but the way the parser borrows its input makes that interface a little tricky.  So ultimately I just decided to remove the `Metta` method altogether.

Possibly in the future we may want to make "Parser" into a trait where we can have a parser that can load code from a file (and start returning results without loading it all into memory at once), and another parser for lambda syntax, etc.  But for now, these are all very far down the priority list.

For now, I've just removed the public API in b2d0314.

Reply:
> Possibly in the future we may want to make "Parser" into a trait where we can have a parser that can load code from a file (and start returning results without loading it all into memory at once), and another parser for lambda syntax, etc. But for now, these are all very far down the priority list.

Yes, it should be probably not even `Parser` but a trait which represents a stream of atoms. Thus the case when user of the API provides just a list of atoms can also be represented by this trait.